### PR TITLE
Add dag to fix metadata

### PR DESCRIPTION
**Pull request by kieranricardo**
This PR adds a dag to fix the metadata. It scrapes the uploaded metadata on s3, finds the matching original metadata from the nci, and copies over the region code and necessary gqa fields before uploading the edited metadata.

This dag was tested in prod, the results of which can be seen in the 2020/2021 metadata files here: http://dea-public-data-dev.s3-ap-southeast-2.amazonaws.com/index.html?prefix=baseline/s2a_ard_granule/

### CODE REVIEW

**Comment by kieranricardo** [comment]
@omad I won't be here on Monday but this should be all ready to auto backfill if you wanna run it

File: scripts/s2_fix_metadata.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
+from odc.aio import S3Fetcher, s3_find_glob
```
Comment (omad) [inline scripts/s2_fix_metadata.py:1]: Same here with the documentation, 
- what will this do, 
- where should it be run (NCI), 
- how (run once per day by the nbart_fix_metadata.py airflow dag)
- what external requrements (libraries which exist in the dea module), 
- and what permissions (AWS creds, as handled by airflow)

File: scripts/s2_fix_metadata.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+    fetcher = S3Fetcher(aws_unsigned=True)
+    s3_obj_stream = s3_find_glob(uri, skip_check=True, s3=fetcher)
+    s3_url_stream = (o.url for o in s3_obj_stream)
+    data_stream = list(fetcher(s3_url_stream))
+    with ThreadPoolExecutor(max_workers=workers) as executor:
+        futures = [executor.submit(process_dataset, s3_obj) for s3_obj in data_stream]
+        
+        for future in as_completed(futures):
+            if future.exception() is not None:
```
Comment (omad) [inline scripts/s2_fix_metadata.py:37]: Not that it matters, but you could probably just call `future.result()` here for the same effect.

**Review by omad — APPROVED** [review]
Looks good, go for it

**Pull request merged**